main.cpp

```diff
@@ -218,10 +226,15 @@
             rtld_fini,
             stack_end
         );
-        if(stateThreadRunning){
-            _DEBUG("Waiting for thread to exit");
-            exitThread = true;
-            stateThread.join();
+        exitThread = true;
+        _DEBUG("Waiting for thread to exit");
+        {
+            std::unique_lock<std::mutex> lock(stateThreadConditionMutex);
+            if(!stateThreadCondition.wait_for(lock, std::chrono::seconds(5), []{
+                return !stateThreadRunning;
+            })){
+                _WARN("Timeout waiting for state thread to exit");
```

Comment:
Could you explain in the context of using the mutex and condition_variable how we would want to keep the locked scope in the lambda passed to the predicate? Especially since stateThreadRunning is only modified by the state thread to indicate if it's finished?